tests/cephfs/clients/client_root_squash_non_root_user.py

```diff
@@ -0,0 +1,240 @@
+import json
+import random
+import secrets
+import string
+import traceback
+
+from ceph.ceph import CommandFailed
+from tests.cephfs.cephfs_utilsV1 import FsUtils
+from utility.log import Log
+
+log = Log(__name__)
+
+
+def validate_fs_info(expected_fs, output):
+    """
+    Validate fs information restriction for clients
+    :param expected_fs - only fs expected to show in output
+    :param output - output of 'ceph fs ls'
+    """
+    if len(output) != 1 and output[0]["name"] != expected_fs:
+        raise CommandFailed("fs is not matching with authorized FS")
+    log.info(f"File systems Information is restricted to {expected_fs} only")
+
+
+def run(ceph_cluster, **kw):
+    """
+     Pre-requisites:
+    - Create a CephFS volume with root squash enabled.
+    - Ensure MDS daemons are deployed for the filesystem.
+    - Set up clients and ensure they are authorized for specific CephFS mounts.
+
+    Test Operations:
+    1. Authorize `client1` for the first CephFS volume with root squash enabled.
+    2. Mount CephFS via both FUSE and kernel mounts on `client1`.
+    3. Run the following scenarios to validate permissions and root squash behavior:
+       - Scenario 1: Attempt to create and delete directories and files as a non-root user. Expecation : should pass
+       - Scenario 2: Attempt to create and delete directories and files as a root user,
```

Comment:
As discussed, scenario2 exists as polarion TC already and is automated. So, can this code block be removed, to avoid redundancy?

Reply:
Hi Suma,
It does not take much time we can include it here as well